scripts/build.py

```diff
@@ -295,29 +295,52 @@ def _compare_file_count(first_dir_path, second_dir_path):
 
 
 def _compare_file_count_for_multiple_dirs(
-        first_dir_list, second_dir_path):
-    """Ensure that two dir's file counts match.
+        first_dir_list, second_dir_list):
+    """Ensure that the total count of files in all directories in the first
+    list matches the count of files in all the directories in the second list.
 
     Args:
        first_dir_list: list(str). List of directories to compare.
-       second_dir_path: str. Second directory to compare.
+       second_dir_list: list(str). List of directories to compare.
 
     Raises:
         ValueError: The source directory list does not have the same file
-            count as the target directory.
+            count as the target directory list.
     """
 
     first_dir_file_count = 0
+    files1 = []
     for first_dir_path in first_dir_list:
         first_dir_file_count += get_file_count(first_dir_path)
-    second_dir_file_count = get_file_count(second_dir_path)
+        for root, _, filenames in os.walk(first_dir_path):
+            for filename in filenames:
+                # Ignore files with certain extensions.
+                filepath = os.path.join(root, filename)
+                if should_file_be_built(filepath) and not any(
+                    filename.endswith(p) for p in FILE_EXTENSIONS_TO_IGNORE):
+                        files1.append(filepath)
+
+    second_dir_file_count = 0
+    files2 = []
+    for second_dir_path in second_dir_list:
```

Comment:
Remove the repetition. See comment above.

Reply:
I have added file_counts but I am not sure how to avoid having two loops for each dir since both have different number of elements for extensions and templates.